src/domain/Game.ts

```diff
@@ -39,4 +43,81 @@ export class Game {
     getWordBeingGuessed() : string {
         return this.wordBeingGuessed;
     }
+
+    isGameOver() : 'won' | 'lost' | 'in-progress' {
+        const maxMissedGuesses = 10;
+        const correctGuessesRequiredToWin = this.getUniqueLettersInAWord();
+
+        if (this.getMissedGuesses() == maxMissedGuesses)
+            return 'lost';
+        else if (this.getCorrectGuesses() == correctGuessesRequiredToWin)
+            return 'won';
+        else
+            return 'in-progress';
+    }
+
+    private getUniqueLettersInAWord() : number {
+        const word = this.getWordBeingGuessed();
+        const wordLetters = Array.from(word);
+
+        return new Set(wordLetters).size;
+    }
+
+    guess(guessedLetter : string) : GuessResponse {
+        if (this.lettersGuessed.includes(guessedLetter))
+            throw new Error("Letter " + guessedLetter + " has already been guessed");
+
+        const wordBeingGuessed = this.getWordBeingGuessed();
+        let gameBuilder;
```

Comment:
gameBuilder = sąlyga
        ? this.updateGameAfterCorrectGuess(guessedLetter);
        :  this.updateGameAfterIncorrectGuess();